overlord/ifacestate/ifacemgr.go

```diff
@@ -77,6 +77,54 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, extraInterfaces
 	return m, nil
 }
 
+func setInitialConnectAttributes(ts *state.Task, plugSnap string, plugName string, slotSnap string, slotName string) error {
+	// Set initial interface attributes for the plug- and slot- snaps in connect task.
+	var snapst snapstate.SnapState
+	var err error
+
+	st := ts.State()
+	if err = snapstate.Get(st, plugSnap, &snapst); err != nil {
+		return err
+	}
+
+	snapInfo, err := snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+
+	plugAttrs := make(map[string]interface{})
+	if plug, ok := snapInfo.Plugs[plugName]; ok {
+		for k, v := range plug.Attrs {
+			plugAttrs[k] = v
```

Comment:
Something smells fishy here. Either it's okay to use the same map, or we need to deep copy the whole thing. If we do need to copy, copying just the first level is not effective.

I suspect we don't need to copy at all, because setting the task state will marshal the value. Let's double check.